### PR TITLE
Card Subtitles - Improve card subtitles for episodes

### PR DESCRIPTION
**Pull request by Camc314**
Wait for #303 

Improves card subtitles for episodes:
If the card is an episode

 - Title: Series Name
 - Subtitle: SX EY - Episode Name

All other movies/series/albums... remain unchanged

### CODE REVIEW

**Comment by MrTimscampi** [comment]
> * Title: Series Name
> * Subtitle: SX EY - Episode Name

Not quite fan of this. The episode name is the main data point here, not the series name.

Something like:
```
SXXEYY - Episode name
Series name
```
Although that puts a LOT of information in a very narrow space, making the episode name essentially useless. Maybe:
```
EYY - Episode name
Series name SXX
```

But neither of them accounts for named seasons, which is something I'd like us to support at some point...

File: components/Card.vue [hunk 1]
```diff
@@ -123,16 +131,19 @@ export default Vue.extend({
           case 'Audio':
           case 'Folder':
           case 'MusicAlbum':
+          case 'MusicArtist':
+          case 'MusicGenre':
```
Comment (dkanada) [inline components/Card.vue:135]: Is there any chance the cards scale to match the height of the sliders now? If that were the case, we could potentially mix items within a single slider for stuff like the search page or the playlist at some point.

Comment (MrTimscampi) [reply components/Card.vue:135]: Technically, nothing prevents us from doing so (Though the cards have different heights, but the slider will take whatever height is the largest).

I don't think making all the cards fit into a 1:1.5 card would look good, though.

**Comment by Camc314** [comment]
I disagree, I think that the Series Name should be at the top (this is done in Apple TV, Plex, and in the current jf-web).

Similarly, in the subtitle, both plex and ATV show the SSXXEYY - Ep name.

@ferferga what do you think

**Comment by ferferga** [comment]
@camc314 I'm trying to see where and how exactly those cards are displayed but I'm not able to, so maybe my opinion is not that great *based on what's currently done*

What I do for most of these design decisions is to check what others do:

* Netflix: In a series page, Netflix shows a list of all the episodes **with the episode's name as title, duration as subtitle and an small overview**.

Related stuff doesn't mention specific episodes, but whole shows

* Prime Video: Amazon show as title "Episode X" and as subtitle the actual episode name. 

Works like Netflix for related stuff.

Wrapping up my opinion: If we're showing an specific episode in a card format and we're sure (or chances are really low) that **another episode of the same show will appear alongside it**, **your format is 100% valid**. Your schema makes sense in those situations where you don't see **the same title repeated over and over**.

However, in general, it seems that we shouldn't recommend episodes unless user is watching the details page of another episode of the same show (couldn't test that as clicking on a episode with this branch merged in master redirects you to playback). So I would only apply your schema there.

Not directly related to the question, but I don't like neither @MrTimscampi and your naming schema: **It doesn't work for localization, as Episode doesn't start with 'E' in all the languages**. We could do something like this:

```
X - Episode Name
Show
```
*(With X starting from **1**, not **01**)*

Ideally, we find a way to place icons or two subtitles, something like this:

```
Episode Name
IconRepresentingSeason X - IconRepresentingNumber Y
```
*(With X and Y being season and episode number respectively)*

**Comment by Camc314** [comment]
> Wrapping up my opinion: If we're showing an specific episode in a card format and we're sure (or chances are really low) that **another episode of the same show will appear alongside it**, **your format is 100% valid**. Your schema makes sense in those situations where you don't see **the same title repeated over and over**.

These cards are only used in the `Up Next` or `Resume Sections`. Consequently, it is unlikely that there will be multiple cards of the same series here. 

The cards in season tabs are different.




> Not directly related to the question, but I don't like neither @MrTimscampi and your naming schema: **It doesn't work for localization, as Episode doesn't start with 'E' in all the languages. Same for seasons**. We could do something like this:

Is there an equivalent to SXXEYY in all languages. If so, we could use translate it to the relevant format in those different languages like how we use tracklist

**Comment by MrTimscampi** [comment]
The best way to go is probably this, then:

Series Title
1. Episode Title

**Comment by ferferga** [comment]
@MrTimscampi Agree with you, I think  we should use Plex's principles in our design as well. What we need to make sure is that server, in case two episodes are added to the same show, only returns us the most recent, so we avoid duplicates, as we discussed earlier. That should be our top priority for this to look nice, it really ruins the experience.

About the localization thing, this is how it looks in Spanish:



*(T equals to 'Temporada')*

I was wondering how we could do this without us needing to look for the first letter of each language (that's why I suggested the icons) but in my mind just popped the idea of doing ``key.charAt(0)`` (``key`` being the translated string for the word "Season" in each language) to solve this problem (I don't know why I didn't think of it before 😅).

If we go that route for localization, we should document this, not only in the code, but also in Weblate (so translators are aware of it). VSCode lets me add comments for each line in the locales, not sure if this would do the trick.

**Comment by MrTimscampi** [comment]
> in case two episodes are added to the same show, only returns us the most recent

It won't ever return two episodes :)

In Next Up, it'll only show the very next episode. In Latest, it'll group them as the series. So we never get more than one.

**Comment by MrTimscampi** [comment]
> I was wondering how we could do this without us needing to look for the first letter of each language (that's why I suggested the icons) but in my mind just popped the idea of doing `key.charAt(0)` (`key` being the translated string for the word "Season" in each language) to solve this problem (I don't know why I didn't think of it before sweat_smile).

Thinking a bit more about this part.  
For non-latin scripts, this approach won't work. I doubt it would fit for Chinese or Japanese, for example. So maybe a dedicated key woul be better. Something like:
```json
{
  "seasonNumberAbbrev": "S{number}",
  "episodeNumberAbbrev": "E{number}"
}
```
This allows all languages to tailor the display for their script and way of doing thing. For example, I'd expect `E10` to be something like  `10話` in Japanese.

**Comment by ferferga** [comment]
@MrTimscampi Yeah, that's great!

**Comment by Camc314** [comment]
I've updated using that suggestion

**Review by MrTimscampi — APPROVED** [review]
LGTM now :)

**Pull request merged**